src/textual/app.py

```diff
@@ -3287,3 +3326,85 @@ def action_command_palette(self) -> None:
         """Show the Textual command palette."""
         if self.use_command_palette and not CommandPalette.is_open(self):
             self.push_screen(CommandPalette(), callback=self.call_next)
+
+    def _suspend_signal(self) -> None:
+        """Signal that the application is being suspended."""
+        self.app_suspend_signal.publish()
+
+    @on(Driver.SignalResume)
```

Comment:
I found the name `SignalResume` confusing here, as I thought this event was directly related to the signalling system. Would `Driver.Resume` suffice?

Reply:
The thing is, it's not a message to indicate that the driver has resumed or that it's been asked to resume; it's a message that is asking the app to publish that (Textual) signal.

Reply:
Shouldn't messages be communicating "a thing happened", while the "what to do with them" side of things is left up to the consumer? When we prescribe an effect in an event/message name it feels like we're enforcing that all consumers must do a certain thing in response, narrowing our options for the future should we ever want to do additional things in response to something.

And if we do want to make it such that only a single behaviour should happen in response to this, perhaps it could be a method instead of an event (events to me imply the potential for multiple consumers).

Reply:
I'll have to dig a bit deeper then, as it wasn't clear to me how to safely call from the driver into the app. What would you suggest as the safe method of performing that?

Reply:
I don't think I'm expressing myself well. I'm trying to understand "why are we soft-enforcing that the only thing we can do in response to an event which says 'the driver has resumed' is publish a signal"? I.e. why is the event named specifically instead of generically to leave open other options for doing things in response to it? Are you intentionally trying to discourage that?

To be clear - this is a genuine question and not a change request. I'm trying to understand this more deeply.

Reply:
Sure, the question makes sense; I'm just trying to work out what the better way is of having the driver call back into the app, to get it to perform a very specific act, without tightly-coupling the driver to the app, and also in such a way as to make sure the act is performed once everything is up and running again.

The issue is that I need `LinuxDriver.start_application_mode` to call [`App._resume_signal`](https://github.com/davep/textual/blob/3a91b3711d5d169e8c06c65fa718a7a6812f4100/src/textual/app.py#L3335-L3337) (so that any code subscribed to [`App.app_resume_signal`](https://github.com/davep/textual/blob/3a91b3711d5d169e8c06c65fa718a7a6812f4100/src/textual/app.py#L605-L613) gets the broadcast at the right moment) under a particular circumstance, but I also don't want to build knowledge about the internals of `App` into `LinuxDriver`. This particular message/event serves no other purpose than to tell `App` that it should broadcast this particular signal.

So what I'm asking is: if not this way, what way would be cleaner?

Reply:
I don't think there's anything wrong with the current the approach at all. I don't find it "unclean". I worry that you think I have much more context than I do, and therefore my genuine clueless questions may be coming across as "you should've done it like this - why didn't you?!".

I'm coming at this with very little context and trying to understand it by asking questions. My context is standup, where I heard you talk about how the SIGABC needs to happen before the SIGBANANA but after the SIGXYZ or your file system will fill up with directories called "zzzcrazybananas", or something like that.

The thing about "perhaps it should be a method" was intended to be a question than a suggestion - but I missed the question mark and unfortunately I can't type with a rising intonation 🤦‍♂️

The only thing I was (poorly) asking/probing about was why the event is named Driver.SignalResume and not Driver.Resume - as it seems to limit how the event _could_ be used in the future and seemed a bit different (much more prescriptive) to how we usually name them.

So that led me to wonder if this event will ever be used for other things - and lead us to want to use the more generic name in the future. For example, would we want to handle the driver resume event and e.g. reload Textual (official) configuration or CSS in the future? Or if we want to add other side effects on "driver resume" in the future inside Textual, should that be done via subscribing to the signal?

Reply:
I don't anticipate there being any other uses for the message; it should very much be considered an internal. I suspect you're right, anything that may wish to react to the resume should probably subscribe to the Signal.

Reply:
Also I wish there was a SIGBANANA.